### PR TITLE
docs: document nested SpatialDiv overflow limitation

### PR DESCRIPTION
**Pull request by walton007**
## Summary

- Document that nested child `SpatialDiv` panels are independent native webview layers and are **not** clipped by parent `overflow: hidden` or parent bounds.
- Add test-server demo `#/nested-spatial-overflow` with spatial vs plain-HTML comparison and toggles for child size and parent overflow.
- Link the limitation from `packages/react/README.md` and maintainer notes in `ARCHITECTURE.md`.

## Test plan

- [ ] Read `docs/webspatial-quirks.md` — *Nested SpatialDiv / parent overflow* section
- [ ] Run `npm run dev`, open `#/nested-spatial-overflow` in visionOS simulator
- [ ] Toggle child size (large > parent) — child spatial panel should remain fully visible past parent bounds
- [ ] Toggle parent `overflow: hidden` — spatial child should stay unclipped; plain HTML reference block should clip as expected

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: packages/react/README.md [hunk 1]
```diff
@@ -26,3 +26,7 @@ Recommended practice: initialize imperative renderers from each container's own
 ## SpatialDiv CSS and transforms
 
 Spatial transforms are sampled from an off-screen **probe** DOM node, not from the visible portal surface. Tag/class selectors, probe vs. host split, and workarounds are documented in [WebSpatial Quirks — SpatialDiv / `enable-xr` and CSS](../../docs/webspatial-quirks.md). Maintainer architecture notes live in [`src/spatialized-container/ARCHITECTURE.md`](src/spatialized-container/ARCHITECTURE.md).
+
+## Nested SpatialDiv overflow
+
+Nested child `SpatialDiv` panels are independent native webview layers. Parent `overflow: hidden` does **not** clip a child `SpatialDiv`; a child larger than its parent will render at full size and can extend past the parent bounds. See [WebSpatial Quirks — Nested SpatialDiv / parent overflow](../../docs/webspatial-quirks.md). Manual check: test-server `#/nested-spatial-overflow`.
```
Comment (khufu-dev) [inline packages/react/README.md]: I don't think we should duplicate this in the Readme. It's redundant.
```suggestion
```

File: packages/react/src/spatialized-container/ARCHITECTURE.md [hunk 2]
```diff
@@ -322,6 +322,18 @@ The probe lives under `cssParserDivContainer` in `<body>`, **not** under the use
...
+Nested child `SpatialDiv` instances are separate native webview layers composited as siblings in the parent's `ZStack` (`Spatialized2DElementView`), not as content inside the parent webview surface. Therefore:
+
+- Parent `overflow` maps to `scrollPageEnabled`, not child clipping.
+- Child size/position come from the child's DOM rect (offset from parent for non-`fixed` children).
+- Parent `materialWithBorderCorner` `clipShape` applies to the parent webview only.
+
+**Workaround:** constrain child CSS size, or use one `SpatialDiv` with plain HTML for overflow clipping inside the portal.
+
+End-user summary: [`docs/webspatial-quirks.md`](../../../../docs/webspatial-quirks.md) — *Nested SpatialDiv / parent overflow*.
+
```
Comment (khufu-dev) [inline packages/react/src/spatialized-container/ARCHITECTURE.md]: I don't think we should duplicate this in the Architecture document. It's actually a bug due to technical limitations not part of the architecture design.
```suggestion
```

**Review by khufu-dev — APPROVED** [review]
